p2p/src/network/pubsub/p2p_network_pubsub_state.rs

```diff
@@ -200,54 +198,203 @@ impl P2pNetworkPubsubClientState {
 
 #[derive(Default, Serialize, Deserialize, Debug, Clone)]
 pub struct P2pNetworkPubsubRecentlyPublishCache {
-    pub map: BTreeSet<Vec<u8>>,
-    pub queue: VecDeque<Vec<u8>>,
+    pub map: BTreeSet<P2pNetworkPubsubMessageCacheId>,
+    pub queue: VecDeque<P2pNetworkPubsubMessageCacheId>,
 }
 
 // TODO: store blocks, snarks and txs separately
 #[derive(Default, Serialize, Deserialize, Debug, Clone)]
 pub struct P2pNetworkPubsubMessageCache {
-    pub map: BTreeMap<Vec<u8>, pb::Message>,
-    pub queue: VecDeque<Vec<u8>>,
+    pub map: BTreeMap<P2pNetworkPubsubMessageCacheId, P2pNetworkPubsubMessageCacheMessage>,
+    pub queue: VecDeque<P2pNetworkPubsubMessageCacheId>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum P2pNetworkPubsubMessageCacheMessage {
+    Init {
+        message: pb::Message,
+        content: GossipNetMessageV2,
+        peer_id: PeerId,
+        time: Timestamp,
+    },
+    PreValidatedBlockMessage {
+        block_hash: mina_p2p_messages::v2::StateHash,
+        message: pb::Message,
+        peer_id: PeerId,
+        time: Timestamp,
+    },
+    // This is temporary handling for transactions and snark pool
+    PreValidated {
+        message: pb::Message,
+        peer_id: PeerId,
+        time: Timestamp,
+    },
+    Rejected {
+        message: pb::Message,
```

Comment:
do we need to keep this if it was rejected? it is not going to be broadcasted or used for anything